keps/77-dynamically-sized-jobs/README.md

```diff
@@ -0,0 +1,286 @@
+  - [Webhook changes](#webhook-changes)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Ignore Resize from Kuberay](#ignore-resize-from-kuberay)
+<!-- /toc -->
+
+## Summary
+
+Enable dynamic sizing of Kueue jobs. For the MVP, we will only focus on supporting RayClusters with autoscaling enabled, but other resources that can benefit from dynamic sizing should be supported eventually.
+
+See: [Support dynamically sized (elastic) jobs #77](https://github.com/kubernetes-sigs/kueue/issues/77)
+
+## Motivation
+
+Kueue currently lacks support for resizing jobs. When a job is resized, Kueue will recreate the Workload representation of the job, leading to a disruptive suspend and requeue process. This limitation hinders the usability of Kueue for resources like RayCluster that sometimes have autoscaling capabilities enabled.
+
+To properly support RayCluster, Kueue needs to gracefully handle the scale up and scale down of RayCluster nodes. Concretely this means that scaling the resources used by a RayCluster is appropriately reflected in the respective ClusterQueue without needing to suspend the entire cluster.
+
+### Goals
+
+- Gracefully handle resize operations for Kueue jobs (i.e. update quota usage without suspend, enqueue scale ups)
+- Autoscaling RayCluster works with Kueue (MVP)
+
+### Non-Goals
+
+- Vertical scaling of workloads – Kueue will only handle resize operations that scale Pods horizontally 
+- Support resize for other Kueue jobs such as QueuedWorkload, JobSet, etc (future)
+- Resizing of RayJobs
+- Partial Preemption
+
+## Proposal
+
+Update the Job framework reconciler and introduce new controllers to orchestrate dynamic resizing of jobs. We are only interested in horizontal scaling of jobs (e.g. scaling more replicas). At a high level, this will be accomplished by:
+- Creating Workload Slice objects that represent incremental scale up of jobs. This gives us per-replica control of workload admission. Workload Slices will be garbage collected and consolidated with their parent workloads after successful admission.
+- Adding default scheduling gates to control the scheduling of new pods based on their admission status.
+- Dynamically adjust quotas in ClusterQueue based on scaling events.
+
+For the MVP, we will only focus on admission of **RayClusters** with autoscaling enabled.
+
+### User Stories
+
+#### Story 1 - RayCluster w/ autoscaling
+
+1. The user creates a RayCluster.
+2. Kueue admits the RayCluster based on the requested resources in the head pod and worker pod. 
+3. User updates RayCluster to enable autoscaling
+4. Kueue will not suspend and requeue the RayCluster, instead it will dynamically update ClusterQueue usage based on the scaling event. 
+
+## Design Details
```

Comment:
I don't see a Nodes/caveats section.

One I can think of is that if Kueue needs to preempt the resized RayCluster, it would preempt it as a whole, regardless of whether the RayCluster was previously scaled up.

Reply:
Okok I added that section.